lib/secrets/secrets.go

```diff
@@ -229,3 +279,155 @@ func PrepareSSHPrivateKey(keyContent string) (string, error) {
 	}
 	return keyPath.Name(), nil
 }
+
+// checks if encryption key exists or not
+func keyExists(basePath string) (bool, error) {
+	path := filepath.Join(basePath, "key")
+	return fileutils.PathExists(path)
+}
+
+func encodeBase64(b []byte) string {
```

Comment:
given that we already use hex encoding in another part of this file, it might make more sense to continue using that encoding

Reply:
Changed to hex now